soh/soh/Enhancements/randomizer/3drando/location_access/locacc_water_temple.cpp

```diff
@@ -20,7 +20,13 @@ void RegionTable_Init_WaterTemple() {
   ---------------------------*/
   if (ctx->GetDungeon(WATER_TEMPLE)->IsVanilla()) {
   //Water Temple logic currently assumes that the locked door leading to the upper water raising location is unlocked from the start
-  areaTable[RR_WATER_TEMPLE_LOBBY] = Region("Water Temple Lobby", "Water Temple", {RA_WATER_TEMPLE}, NO_DAY_NIGHT_CYCLE, {}, {}, {
+  areaTable[RR_WATER_TEMPLE_LOBBY] = Region("Water Temple Lobby", "Water Temple", {RA_WATER_TEMPLE}, NO_DAY_NIGHT_CYCLE, {}, {
+                  //Locations
+                  LOCATION(RC_WATER_TEMPLE_MAIN_LEVEL_2_POT_1, logic->CanBreakPots()),
+                  LOCATION(RC_WATER_TEMPLE_MAIN_LEVEL_2_POT_2, logic->CanBreakPots()),
+                  LOCATION(RC_WATER_TEMPLE_MAIN_LEVEL_1_POT_1, logic->CanBreakPots()),
+                  LOCATION(RC_WATER_TEMPLE_MAIN_LEVEL_1_POT_2, logic->CanBreakPots()),
```

Comment:
Don't these need water level checks? I assume level 1 is the bottom floor?

Reply:
[23-52-80-42-45.json](https://github.com/user-attachments/files/18045214/23-52-80-42-45.json)
Attached spoiler log of affected playthrough. Requires I get the wallet in RC_WATER_TEMPLE_MAIN_LEVEL_1_POT_1 to buy the A button for the ocarina from the shop, to play the song of time, to become an adult. Can't get to RC_WATER_TEMPLE_MAIN_LEVEL_1_POT_1 without being an adult.

Reply:
Even weirder, Water Temple was behind GTG gate, and that access was in sphere 5, whereas the pot was shown as accessible in sphere 2, so it was doubly adult-locked.

Reply:
While they (and a lot of other stuff) could use better names, The level 1 pots are the ones near the boss door, and the level 2 pots are on the middle level, on the way to the singers spikes area.

This means that the level 1 pots need high level + longshot (or more accurately, moving to RR_WATER_TEMPLE_PRE_BOSS_ROOM), and the level 2 pots need irons + hookshot or water level (low or mid).